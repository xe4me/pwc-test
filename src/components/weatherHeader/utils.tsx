import format from 'date-fns/format';

export const formatTodayDate = () => format(new Date(), 'cccc, MMMM do');
