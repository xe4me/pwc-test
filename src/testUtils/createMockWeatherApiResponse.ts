export const createMockWeatherApiResponse = (city_name: string) => ({
    city_name,
    data: [
      {
        wind_cdir: 'W',
        pres: 1014,
        high_temp: 20.2,
        wind_gust_spd: 5.9,
        wind_spd: 3.1,
        dewpt: 5.9,
        weather: {
          icon: 'c02d',
          code: 801,
          description: 'Few clouds',
        },
        wind_dir: 266,
        precip: 0,
        low_temp: 9.3,
        max_temp: 20.2,
        datetime: '2022-06-25',
        temp: 14.9,
        min_temp: 9.9,
      },
    ],
  });
