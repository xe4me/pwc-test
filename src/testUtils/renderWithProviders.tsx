import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material';
import theme from '../theme';

export const renderWithProviders = (element: ReactElement) =>
  render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
