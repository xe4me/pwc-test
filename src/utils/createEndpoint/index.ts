import { envs } from 'constants/envs';

/*
 * @Function createEndpoint
 * This function will add the BASE URL and the authentication key to the given endpoint
 * Note that this function is very limited and does not consider whether the endpoint has "$" or not
 * */
export const createEndpoint = (endpoint: string) =>
  `${envs.weatherApiBaseUrl}${endpoint}&key=${envs.authKey}`;
