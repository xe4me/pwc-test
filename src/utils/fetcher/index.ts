/*
 * @Function fetcher
 * This is a wrapper around the browser's fetch function, required by SNW library
 * */
// @ts-ignore
export const fetcher = (...args) => fetch(...args).then((res) => res.json());
