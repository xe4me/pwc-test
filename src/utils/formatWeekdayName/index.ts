import format from 'date-fns/format';
import isToday from 'date-fns/isToday';

/*
* @Function formatWeekdayName
* Will format the given date to one of these values
* [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ]
  * If the given date is current day, it will return Today
* */
export const formatWeekdayName = (datetime: Date) =>
  isToday(datetime) ? 'Today' : format(datetime, 'cccc');
