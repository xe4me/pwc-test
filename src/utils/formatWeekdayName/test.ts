import { addDays, subDays } from 'date-fns';
import { formatWeekdayName } from './index';

describe('formatWeekdayName()', () => {
  it('should return Today, if the date is current day', () => {
    expect(formatWeekdayName(new Date())).toBe('Today');
  });
  it('should return the correct name of day, parsed from the given date', () => {
    const date = new Date();
    const days = [
      'Sunday',
      'Monday',
      'Tuesday',
      'Wednesday',
      'Thursday',
      'Friday',
      'Saturday',
    ];
    const currentDay = date.getDay();
    let testDate;
    let expectedDay;
    if (currentDay === 6) {
      testDate = subDays(new Date(), 1);
      expectedDay = days[currentDay - 1];
    } else {
      testDate = addDays(new Date(), 1);
      expectedDay = days[currentDay + 1];
    }
    expect(formatWeekdayName(testDate)).toBe(expectedDay);
  });
});
