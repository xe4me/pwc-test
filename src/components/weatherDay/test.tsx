import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithProviders } from 'testUtils/renderWithProviders';
import { axe } from 'jest-axe';
import { formatWeekdayName } from 'utils/formatWeekdayName';
import { WeatherDay } from './index';

const renderComponent = () => renderWithProviders(
    <WeatherDay datetime={new Date().toISOString()} />
  );
describe('<WeatherDay>', () => {
  describe('Snapshot and accessibility', () => {
    it('should match with the snapshot', () => {
      const { baseElement } = renderComponent();
      expect(baseElement).toMatchSnapshot();
    });
    it('should pass accessibility tests', async () => {
      const html = renderComponent();
      expect(await axe(html.container)).toHaveNoViolations();
    });
  });
  describe('Visible Elements', () => {
    it('should render the formatted given date', () => {
      renderComponent();
      const formattedDate = formatWeekdayName(new Date());
      expect(screen.queryByText(formattedDate)).toBeInTheDocument();
    });
  });
});
