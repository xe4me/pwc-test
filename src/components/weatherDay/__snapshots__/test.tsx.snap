// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDay> Snapshot and accessibility should match with the snapshot 1`] = `
<body>
  <div>
    <h4
      class="MuiTypography-root MuiTypography-subtitle1 css-1isisa7-MuiTypography-root"
    >
      Today
    </h4>
  </div>
</body>
`;
